Adding Song Smells Like Teen Spirit

DeepSix opened the pull request.

jujudusud commented:
fixes #5

theGreatWhiteShark reviewed (commented):
Hey,

Could you also add a **license** to the song and pattern files? This clarifies for other people if and how they can use those files (and without a permissive one we wouldn't be allowed to host it).

Some common licenses can be found here http://hydrogen-music.org/documentation/manual/manual_en.html#chpt.license.common_licenses

DeepSix commented:
Added to the PR to include the CC LICENSE file.    Please verify this is what is missing.

theGreatWhiteShark commented:
> Added to the PR to include the CC LICENSE file. Please verify this is what is missing.

Hey, thanks. But that's not what I meant.

Both `.h2song` and `.h2pattern` have a `<license></license>` element where the license should be entered. You can do so in Hydrogen itself using the song properties dialog from the main menu and the pattern properties dialogs when right clicking each of them. Or you can just replace the `Unknown license` with `CC0` in the commited files.

Defining the license per file is required as people may just pick some patterns instead of the whole song/folder. Maybe there will be some sort of online pattern/song repo similar to the way drumkits are shared right now where people can decide to only pick specific files.

The `.autosave` backup file of the song is not required either.

DeepSix commented:
Apologies for my ignorance on the Lic.     I understand now.    I added CC0   Free to use  in the song info - license field.    I removed autosave, removed the extra patterns that were redundant, and added a new song to the PR.    I appreciate your help and patience.   Hope I have it right this time.

theGreatWhiteShark commented:
> Apologies for my ignorance on the Lic. I understand now. I added CC0 Free to use in the song info - license field. I removed autosave, removed the extra patterns that were redundant, and added a new song to the PR. I appreciate your help and patience. Hope I have it right this time.

Don't worry.

The license in the ZZ Top song is fine but it is still missing in the Nirvana song. (BTW I noticed your still using version 1.1.1 of Hydrogen. Using the [latest](https://github.com/hydrogen-music/hydrogen/releases/download/1.2.2/Hydrogen-1.2.2-x86_64.AppImage) version you can set the license using a dropdown in Main Menu > Project > Song Properties).

Just a remark (nothing you need to fix): the kick drum is so loud it is clipping (reaches so high the wave form is cut at the highest value). Instead of making the kick drum louder you could decrease volume of all other instruments while making the master volume louder.

DeepSix commented:
Corrected Nirvana, Added immigrant song to the PR.    
I'll add a github 'linting' check to make sure I have the CC0 as a lic for any new songs in case I forget.   
I hope these songs are of value....   If they are, I'll keep adding new ones.   I use them to practice guitar, so the patterns are more important than the sounds for my use.    

Any requests?

theGreatWhiteShark commented:
> I hope these songs are of value

Sure. For now they will be little hard to discover for other Hydrogen users, though.

But I'm currently working on a feature that will allow switching drumkits without requiring the user to fix pattern and importing arbitrary patterns in a song. With this is place in version 1.3 sharing and hosting songs and patterns will move into focus again and it will definitely help to already have a set of songs to offer.

> I use them to practice guitar, so the patterns are more important than the sounds for my use.

That's totally fine. And since they are licensed as CC0 fine-tuning could be done by a different person.

> Any requests?

Hmm. Last song I played along was Spilling Holy Blood by Profanatica. But that's probably interesting for way less people than a Led Zeppelin song is :laughing: .
